src/VirtoCommerce.OrdersModule.Web/Controllers/Api/OrderModuleShipmentsController.cs

```diff
@@ -0,0 +1,42 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using VirtoCommerce.OrdersModule.Core;
+using VirtoCommerce.OrdersModule.Core.Model;
+using VirtoCommerce.OrdersModule.Core.Services;
+using VirtoCommerce.OrdersModule.Data.Authorization;
+
+namespace VirtoCommerce.OrdersModule.Web.Controllers.Api
+{
+    [Route("api/order/shipments")]
+    [Authorize]
+    public class OrderModuleShipmentsController : Controller
+    {
+        private readonly IShipmentService _shipmentService;
+        private readonly IAuthorizationService _authorizationService;
+
+        public OrderModuleShipmentsController(IShipmentService shipmentService,
+            IAuthorizationService authorizationService)
+        {
+            _shipmentService = shipmentService;
+            _authorizationService = authorizationService;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> UpdateShipment([FromBody] Shipment shipment)
+        {
+            var authorizationResult = await _authorizationService.AuthorizeAsync(
+                User,
+                null,
+                new OrderAuthorizationRequirement(ModuleConstants.Security.Permissions.UpdateShipments));
+            if (!authorizationResult.Succeeded)
+            {
+                return Unauthorized();
+            }
```

Comment:
Add captions to arguments, please. We need to clarify what is `null` here for better readability.
```suggestion
            var authorizationResult = await _authorizationService.AuthorizeAsync(
                User,
                null,
                new OrderAuthorizationRequirement(ModuleConstants.Security.Permissions.UpdateShipments));
            
            if (!authorizationResult.Succeeded)
            {
                return Unauthorized();
            }
```